src/components/Select/Select.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { Ref, FC, useState, useEffect } from 'react';
+
+// Styles:
+import { Dropdown, ToggleState } from '../Dropdown';
+import { ButtonIconAlign, DefaultButton } from '../Button';
+import { IconName } from '../Icon';
+import { Props } from '@storybook/addon-docs';
+import { Menu } from '../Menu';
+import { SearchBox, TextInput } from '../Inputs';
+import { SelectOption, SelectProps } from './Select.types';
+import { Pill, PillSize } from '../Pills';
+
+const Options = (options: SelectOption[], onOptionChange: Function) => {
+    return (
+        <Menu
+            items={options}
+            onChange={(value) => {
+                const option = options.find((option) => option.value === value);
+                onOptionChange(option);
+            }}
+        />
+    );
+};
+
+export const Select: FC<SelectProps> = React.forwardRef(
+    ({ options = [], ...rest }, _ref: Ref<HTMLButtonElement>) => {
+        const {
+            defaultValue,
+            disabled,
+            clearable = false,
+            filterable = false,
+            multiple = false,
+            loadOptions,
+            isLoading = false,
+        } = rest;
```

Comment:
Recommend looking at other component patterns to narrow the scope of this spread using `OcBaseProps`. Would like to hear your thoughts on this implementation.